buildenv/jenkins/jobs/infrastructure/copyrightCheck

```diff
@@ -61,30 +68,313 @@ timeout(time: 6, unit: 'HOURS') {
+                                            if (foundIBMCopyright < foundOracleCopyright ||
+                                                foundIBMCopyright < foundBSDorMITCopyright ||
+                                                foundIBMCopyright < foundApacheCopyright ||
+                                                foundIBMCopyright < foundCopyright) {
+                                                    //The IBM Copyright is not after the existing copyright in the file
+                                                    errorStr = "IBM Copyright is not after the existing copyright"
+                                                    FAIL = addError(BAD_FILES, "${file}", errorStr)
+                                            }
+                                        } else {
+                                            errorStr = "IBM Copyright (basic) is missing from the file"
+                                            FAIL = addError(BAD_FILES, "${file}", errorStr)
+                                            checkCopyrightDate = false
+                                        }
+                                    }
+                                } // !foundNonIBMCopyright
+                            } // if closed
                         } else {
-                            echo "Copyright date in file: appears to be correct"
+                            // must be an eclipse repository, so just 
+                            // check that the date is valid
+                            checkCopyrightDate = true
+                            println "Checking file: '${file}'"
+                        }
+                            
+                         // We need to check that the copyright update year is the current year 
+                        if (checkCopyrightDate) {
+                            RESULT = sh (
+                                script: "grep -qE ${REGEX} '${file}'",
+                                returnStatus: true)
+                            if(RESULT != 0) {
+                                errorStr = "IBM Copyright date appears to be incorrect"
+                                FAIL = addError(BAD_FILES, "${file}", errorStr)
+                            } else {
+                                echo "Copyright date in file: appears to be correct"
+                            }
                         }
                     }
+                    
                     if (FAIL) {
                         echo "${HASHES}"
                         echo "The following files were modified and have incorrect copyrights"
-                        BAD_FILES.each() {
-                            echo "${it}"
+                        BAD_FILES.keySet().each {badfile ->
+                            echo "${badfile}"+BAD_FILES.get("${badfile}")
                         }
                         echo "${HASHES}"
-                        sh 'exit 1'
+                        currentBuild.result = 'FAILURE'
```

Comment:
I think this one is fine because its the last line of code.

Reply:
OK, set back to original